client/pluginmanager/csimanager/instance.go

```diff
@@ -73,12 +75,7 @@ func newInstanceManager(logger hclog.Logger, eventer TriggerNodeEvent, updater U
 }
 
 func (i *instanceManager) run() {
-	c, err := csi.NewClient(i.info.ConnectionInfo.SocketPath, i.logger)
-	if err != nil {
-		i.logger.Error("failed to setup instance manager client", "error", err)
-		close(i.shutdownCh)
-		return
-	}
+	c := i.newClient(i.shutdownCtx, i.info.ConnectionInfo.SocketPath)
```

Comment:
Currently this is causing some unit tests to hang because we previously would throw an error and exit, whereas now we wait forever for the socket to start up. Probably will need to use the context that `newInstanceManager` creates to thread down to here.